server/services/nl.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Language Model client."""
+
+from google.cloud import storage
+from sentence_transformers import SentenceTransformer
+from sentence_transformers.util import semantic_search
+
+import os
+import torch
+from datasets import load_dataset
+import logging
+
+BUILDS = ['demographics300', 'uncurated3000']
+BUILD = 'uncurated3000'  #@param ['demographics300', 'uncurated3000']
+QUERY = "people who cannot see"  #@param {type:"string"}
+GCS_BUCKET = 'datcom-csv'
+EMBEDDINGS = 'embeddings/'
+TEMP_DIR = '/tmp/'
+
+
+class Model:
+  """Holds clients for the language model"""
+
+  def __init__(self):
+    self.model = SentenceTransformer('all-MiniLM-L6-v2')
+    self.dataset_embeddings_maps = {}
+    self.download_embeddings()
+    self.dcid_maps = {}
+    for build in BUILDS:
+      logging.info('Loading build {}'.format(build))
+      ds = load_dataset('csv', data_files=f'/tmp/embeddings_{build}.csv')
+      df = ds["train"].to_pandas()
```

Comment:
To confirm, this is using pandas, is that okay?

Also, I hope numpy is okay to include because many of the libs might depend on it